core/invoker/src/main/scala/org/apache/openwhisk/core/containerpool/ContainerProxy.scala

```diff
@@ -560,7 +560,9 @@ class ContainerProxy(
         }
         val parameters = job.msg.content getOrElse JsObject.empty
 
-        val authEnvironment = job.msg.user.authkey.toEnvironment
+        val authEnvironment = if (job.action.annotations.isTruthy(WhiskAction.provideApiKeyAnnotationName)) {
+          job.msg.user.authkey.toEnvironment
+        } else JsObject.empty
```

Comment:
To avoid a breaking change: Can we assume the absence of the annotation to mean "true" and create all new actions with the annotation set to false?

Reply:
possibly brilliant - yes if the annotation is added server side... will munch on the suggestion - thanks!!

Reply:
implemented this proposal - thanks @markusthoemmes. Can you review the changes now 🙏 ?

Reply:
just want to confirm my understanding as well as the non-breaking behavior in an update scenario.  from the dev list discussion
> actions which do not have the annotation will still get the api key injected. The annotation is added by the controller when an action is created or updated unless already present. The default value for the annotation is "false", meaning no key is injected to the action context.
here's the scenario
1. an existing action without annotation is updated, the request does not specify the annotation
2. backend sees action without annotation and updates action with the annotation
    - will this value be `true` or `false` ?
3. some time later, another request to update the same action is made.  as before, the request does not specify the annotation.
    - will the updated action have the annotation set to `true` or `false`?

Reply:
As implemented here, the value for the annotation when added by the system is always `false`. The first update will add the annotation. So in the scenario above, the second update would already have the annotation and the system leaves it in place.

Reply:
The tests here are for what I describe above:

https://github.com/apache/incubator-openwhisk/pull/4284/files#diff-497ffa2cac9fe692a1df3b53e9793a41R1677

https://github.com/apache/incubator-openwhisk/pull/4284/files#diff-497ffa2cac9fe692a1df3b53e9793a41R1686

Reply:
Thanks @mdeuser for taking a look at this proposal.

Reply:
for the most backwards compatibility, i'm thinking trying to preserve the behavior for existing actions unless explicitly changed by the action owner.

as such, updates to existing actions would have the annotation set to `true` (i.e. retain __ow_api_key behavior) if the annotation did not exist and the annotation is not specified on the request.  if the annotation exists and the annotation was not specified on the update request, retain the annotation existing value.

newly created actions would get the default annotation value of `false` if not specified on the request.

Reply:
> as such, updates to existing actions would have the annotation set to `true`

what if the update request is updating unrelated annotations -- eg to make an action a web action or modifying descriptions?

Reply:
let's see.  that's an update to an existing action.  so, if the action did not have the api_key annotation, then add it and set it to `true`.  if the action did have the annotation, leave it as-is.

Reply:
take a look at the last commit @mdeuser - implemented your suggestions.

Reply:
LGTM.  thanks!